salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -52,13 +59,15 @@ Run the highstate:
   salt.state:
     - tgt: {{ pillar['node_name'] }}
     - highstate: True
+    - pillar: {{ pillar_data | tojson }}
     - require:
       - salt: Set grains
       - salt: Refresh the mine
 
 {%- if 'etcd' in pillar.get('metalk8s', {}).get('nodes', {}).get(pillar['node_name'], {}).get('roles', []) %}
 
-  {%- set node_ip = grains['metalk8s']['control_plane_ip'] %}
+  {#- Bug: at this point, the mine may not be filled yet because we render the SLS on the master first #}
```

Comment:
Can't we ensure it is filled by splitting the orchestrate or calling some `mine.update` first?

Reply:
We should basically not use the mine for this we have a ticket #877.
Currently I will fix it by retrieving mine information directly in the module if we have no endpoint